src/cargo/util/toml/mod.rs

```diff
@@ -2144,6 +2144,8 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
                 .unwrap_or(GitReference::DefaultBranch);
             let loc = git.into_url()?;
 
+            bail_if_github_pull_request(&name_in_toml, &loc)?;
```

Comment:
I'm a bit uncomfortable proactively erroring if the URL pattern matches to github, rather than providing context if what the user specified fails.  I guess the worst that can happen is github allows you to clone PR URLs and we people report it and we have to remove it to allow its use.  Not the worst outcome for something that is unclear if it would ever happen.

Reply:
The behavior here is the same as for how fragments are handled directly below this (though they are handled as a warning. The difference is that this causes an error later in the processing while the fragment does not. I think it's reasonable to avoid attempting to do cause an error by downloading. If this was wrong, then removing this code or changing it to a warning would be fairly low cost. How about we keep this as-is?

Reply:
A warning is safer if we are unsure about the effect of a change. I would vote for turning into a warning for now :)

Reply:
Will fix

Reply:
Done